Cython/Compiler/ModuleNode.py

```diff
@@ -964,9 +964,11 @@ def generate_filename_table(self, code):
             for source_desc in code.globalstate.filename_list:
                 file_path = source_desc.get_filenametable_entry()
                 if isabs(file_path):
-                    file_path = basename(file_path)  # never include absolute paths
-                escaped_filename = file_path.replace("\\", "\\\\").replace('"', r'\"')
-                escaped_filename = as_encoded_filename(escaped_filename)
+                    # never include absolute paths
+                    file_path = source_desc.get_description()
+                # Always use / as separator
+                file_path = file_path.replace('\\', '/')
+                escaped_filename = as_encoded_filename(file_path)
                 code.putln('%s,' % escaped_filename.as_c_string_literal())
```

Comment:
I'm not sure whether we should only make paths relative on output or rather earlier on input. We do put some effort into making paths absolute and comparable, though, so maybe absolute paths are still better for finding and processing files.

If we only do this on output, then we should move the "make relative" logic into `source_desc.get_filenametable_entry()` (see `Scanning.py`). It should consistently return the same file path usable for putting it in the filename table (which is what we do here), without further adaptation (except for C output escaping).